include/cuco/detail/static_map/static_map_ref.inl

```diff
@@ -585,14 +586,167 @@ class operator_impl<
       cuda::std::is_invocable_v<Op, cuda::atomic_ref<T, Scope>, T>,
       "insert_or_apply expects `Op` to be a callable as `Op(cuda::atomic_ref<T, Scope>, T)`");
 
+    auto& ref_ = static_cast<ref_type&>(*this);
+
+    // directly dispatch to implementation if no init element is given
+    auto constexpr init_eq_sentinel = false;
+    ref_.insert_or_apply_impl<init_eq_sentinel>(value, op);
+  }
+
+  template <typename Value>
+  __device__ void insert_or_apply(Value const& value, cuco::op::reduce::sum_tag)
+  {
+    auto& ref_ = static_cast<ref_type&>(*this);
+    ref_.insert_or_apply(value, T{}, [](cuda::atomic_ref<T, Scope> payload_ref, T const& payload) {
```

Comment:
>if no init value is provided

The `cuco::op::reduce::*` overloads do not take `init` as of now. As we know the type of **binary operation** we just initalize the **init** value by ourself. For example : **sum_tag** will use 0, **min_tag** will `std::numeric_limits<T>::max()`, and **max_tag** will use `std::numeric_limits<T>::min()`. 


>  the default value of the payload will become the init?

We cannot be sure. If the default value of the payload is not the actual init value of the binary operator, then it would give wrong results from the reduction. Hence, as we know the binary operation before hand, we can be sure to initalize with correct one.